### ISSUE TITLE
Incremental SSG doesnt return data randomly

### ISSUE THREAD

**Issue by nimitwalia89** [vercel/next.js#16409]
# Bug report

## Describe the bug

On some ssg pages with fallback and revalidate it doesnt return any data for some reason. Either params are not being sent or api fails. We tried refreshing multiple times and nothing happens. But as soon as we restart next js server it starts working after 2 refreshes. 

Its been happening since we migrated to ssg from ssr.
Pages work at first. Then stops randomly and article not found is shown which only comes wen no data is returned.

Please help.

## To Reproduce

It just happens randomly. Maybe because its a big website with lots of articles. 

## Expected behavior

Pages should always be available.

## System information

- OS: Any
- Browser: any
- Version of Next.js: 9.5.2

**Comment by Timer** [comment]
Can you please share your code?

**Comment by nimitwalia89** [comment]
Sure.

`
export default function Home({ article, params, domain, errorCode }) {
...
}
export async function getStaticPaths() {
  return { paths: [], fallback: true };
}

export const getStaticProps = async ({ params }) => {
  const query = `some query`;
  const { slug, language, site } = params;
  const domain = config.domains.find((item) => item.language === language);
  const siteDomain = domain?.sites.find((item) => item.site === site) || null;
  try {
    const data = await fetchGraphqlAPI(query, { slug, language });
    return {
      props: {
        article: data?.postBySlug || null,
        params,
        domain: siteDomain,
        errorCode: !siteDomain ? 404 : false,
      },
      revalidate: 15,
    };
  } catch (err) {
    console.log("error", err);
    return {
      props: { params, domain },
    };
  }
};`

**Comment by Timer** [comment]
You're catching errors and returning props causing a render without your data.

This will work if you stop swallowing the error:

```tsx
export const getStaticProps = async ({ params }) => {
  const query = /* some query */ {};
  const { slug, language, site } = params;
  const domain = config.domains.find((item) => item.language === language);
  const siteDomain = domain?.sites.find((item) => item.site === site) || null;

  const data = await fetchGraphqlAPI(query, { slug, language });
  return {
    props: {
      article: data?.postBySlug || null,
      params,
      domain: siteDomain,
      errorCode: !siteDomain ? 404 : false,
    },
    revalidate: 15,
  };
};
```

**Comment by nimitwalia89** [comment]
What i dont understand is why does it work only after i restart the server. i ll give it a try to avoid catching error

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
